src/http_tracker_connection.cpp

```diff
@@ -272,6 +272,20 @@ namespace libtorrent {
 	void http_tracker_connection::on_filter(http_connection& c
 		, std::vector<tcp::endpoint>& endpoints)
 	{
+		// filter all endpoints we cannot reach from this listen socket, which may
+		// be all of them, in which case we should not announce this listen socket
+		// to this tracker
+		auto const ls = bind_socket();
+		endpoints.erase(std::remove_if(endpoints.begin(), endpoints.end()
+			, [&](tcp::endpoint const& ep) { return !ls.can_route(ep.address()); })
+			, endpoints.end());
+
+		if (endpoints.empty())
+		{
+			fail(error_code(boost::system::errc::host_unreachable, system_category()));
+			return;
+		}
```

Comment:
There's similar code in `http_connection::on_resolve` which should probably be folded into this.